### ISSUE TITLE
streamingccl: TestTenantStreamingFailback timed out

### ISSUE THREAD

**Issue by adityamaru** [cockroachdb/cockroach#112554]
Seen in https://teamcity.cockroachdb.com/viewLog.html?buildId=12221602&buildTypeId=Cockroach_BazelExtendedCi

Jira issue: CRDB-32491

**Comment by msbutler** [comment]
Intriguing. I'll note that I was recently looking at a test failure (on 23.1) where shutdown was stuck potentially by a gc finalizer that we could not observe in the stacks in the artifacts.
https://github.com/cockroachdb/cockroach/issues/112469#issuecomment-1768749191

**Comment by adityamaru** [comment]
I'm going to err on the side of closing this out. If we see it again or get dinged for it we can revisit.

**Comment by DrewKimball** [comment]
I saw it again here: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/12330786?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildTestsSection=true

**Comment by rafiss** [comment]
It flaked (no timeout) here: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/12491945?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true

```
    stream_ingestion_job_test.go:744: tenant "g" is still accepting connections
    stream_ingestion_job_test.go:747: tenant "g" is not accepting connections
    stream_ingestion_job_test.go:285: starting replication f->g
    stream_ingestion_job_test.go:289: waiting for g@1699036101656260960.0000000000
    stream_ingestion_job_test.go:168:
          Error Trace:  github.com/cockroachdb/cockroach/pkg/ccl/streamingccl/streamingest/stream_ingestion_job_test.go:168
                              github.com/cockroachdb/cockroach/pkg/ccl/streamingccl/streamingest/stream_ingestion_job_test.go:299
          Error:        Not equal:
                        expected: 4438609793771831563
                        actual  : -5645868812438106101
          Test:         TestTenantStreamingFailback
          Messages:     fingerprint mismatch at 1699036101656260960.0000000000
```

**Comment by stevendanna** [comment]
Yikes. A checksum mismatch. Thanks for reporting this!

**Comment by stevendanna** [comment]
From reading the logs, I think this is a race in the test. Namely, we currently don't have a great way to make sure the tenant is actually shut down. From the logs, it appears that the tenant may have still been up at the point that we tried to reset its timestamp.

**Comment by dt** [comment]
I'm demoting this from GA blocker to p-2 since tailback isn't officially supported in 23.2 anyway and this is just a race condition that a runbook could handle with a delay.

**Comment by dt** [comment]
Fixed in master and unsupported in 23.2. Closing. If this recurs, we'll delete the 23.2 test.
